.github/workflows/build_and_test.yaml

```diff
@@ -42,6 +42,7 @@ jobs:
         run: mvn -B -DskipTests package
       - name: Test
         run: mvn -B test
-      # Optional: Uploads the full dependency graph to GitHub to improve the quality of Dependabot alerts this repository can receive
-      - name: Update dependency graph
-        uses: advanced-security/maven-dependency-submission-action@571e99aab1055c2e71a1e2309b9691de18d6b7d6
+      - name: Submit Dependency Snapshot
```

Comment:
I don't think this is necessary for every pull request. It should be only done on push.